src/sql/db_connection_pool/mod.rs

```diff
@@ -48,3 +49,22 @@ impl From<&str> for Mode {
         }
     }
 }
+
+/// A key that uniquely identifies a database instance.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum DbInstanceKey {
+    /// The database is a file on disk, with the given path.
+    File(Arc<str>),
+    /// The database is in memory.
```

Comment:
nit: possibly worth mentioning that we explicitly only want to allow a single in-memory db.

Reply:
I'll do this in a follow-up PR for SQLite